CODEOWNERS

```diff
@@ -1,3 +1,12 @@
+# require @mozilla/data-platform-infra-wg review for everything else
+* @mozilla/data-platform-infra-wg
+
+# do not require review from data platform for the following files and directories
+/sql/
+/sql_generators/
+/tests/sql/
+dags.yaml
```

Comment:
Based on the official docs, this will exclude the listed directories from requiring review as no codeowner has been specified for those: https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-code-owners#example-of-a-codeowners-file

Reply:
Also, maybe we should require DE review for `dataset_metadata.yaml` files?  Otherwise restricted access datasets could be reconfigured without our knowledge, and there are conventions for dataset names and ACLs that we need to make sure are followed consistently.

While adding new datasets and changing dataset permissions isn't uncommon, it doesn't seem super frequent, so I'm thinking requiring DE review for adding/changing datasets would hopefully not be too much of a bottleneck.

Reply:
Yeah, that sounds like a good idea